trip_app/accounts/permissions.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework import permissions
+
+
+class IsAuthOwnerOrAdminOnly(permissions.IsAuthenticated):
+    """
+    Custom permission to only allow owners and admins to get/edit it.
+    """
+
+    def has_object_permission(self, request, view, obj):
+        # Write permissions are only allowed to the owner or admin
+        is_auth = super().has_object_permission(request, view, obj)
+        return (is_auth and obj == request.user) or (is_auth and request.user.is_superuser)
```

Comment:
seems like you can:
1. return false if `not is_auth`
1. have simpler `or` for your rules

Reply:
Yeah, agree with both points... I will address this in coming PR, if you don't mind.